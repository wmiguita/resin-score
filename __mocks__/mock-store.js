import configureStore from "redux-mock-store"

const mockStore = configureStore([])

//TODO: find better way to keep reducers up to date and check for any test side effects
export const store = mockStore({
  atheletes: {
  },
  userInterface: {
    drawerOpen: false,
    selectedTab: 0
  }
})
