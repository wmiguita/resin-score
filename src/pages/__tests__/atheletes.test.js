import React from "react"
import { Provider } from "react-redux"
import renderer from "react-test-renderer"

import { wrapComponent } from "mock-store"
import AtheletesPage from "../atheletes"

describe( "Atheletes page", () => {
  it( "renders athelete page on listing tab", () => {
    const page = renderer.create( wrapComponent( <AtheletesPage /> )).toJSON()

    expect( page ).toMatchSnapshot()
  });
})
