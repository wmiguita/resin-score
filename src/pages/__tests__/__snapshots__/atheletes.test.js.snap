// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atheletes page renders athelete page on listing tab 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h2
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h4"
    >
      Athelete list
      <button
        aria-label="add"
        className="MuiButtonBase-root MuiFab-root makeStyles-addAction-6 MuiFab-sizeSmall MuiFab-primary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
      </button>
    </h2>
    <ul
      className="MuiList-root MuiList-padding"
    />
  </div>
</div>
`;

exports[`Atheletes page renders athelete page on manage tab 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h2
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h4"
    >
      Athelete list
      <button
        aria-label="add"
        className="MuiButtonBase-root MuiFab-root makeStyles-addAction-6 MuiFab-sizeSmall MuiFab-primary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </h2>
    <ul
      className="MuiList-root MuiList-padding"
    />
  </div>
</div>
`;
